skyportal/handlers/api/hermes/hermes.py

```diff
@@ -0,0 +1,240 @@
+import functools
+import json
+
+import requests
+
+from baselayer.app.access import auth_or_token
+from baselayer.app.env import load_env
+from baselayer.app.flow import Flow
+from baselayer.app.handlers.base import BaseHandler
+from baselayer.log import make_log
+
+from ....models import TNSRobot
+from ....utils.data_access import get_publishable_obj_photometry
+
+_, cfg = load_env()
+log = make_log("api/hermes")
+
+
+def catch_timeout_and_no_endpoint(func):
+    """
+    Catch timeout and missing endpoint errors from the Hermes server
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except requests.exceptions.Timeout:
+            raise ValueError("Unable to reach the Hermes server")
+        except KeyError as e:
+            if "endpoint" in str(e):
+                raise ValueError("Hermes endpoint is missing from configuration")
+            else:
+                raise e
+
+    return wrapper
+
+
+def create_payload_and_header(obj, photometry, data):
+    """
+    Create the payload qnd the header for Hermes and validate it by using the Hermes API
+    """
+    header = {
+        "Authorization": f"Token {cfg['app.hermes.token']}",
+        "Content-Type": "application/json",
+    }
+
+    payload = {
+        "topic": cfg["app.hermes.topic"],
+        "title": data["title"],
+        "submitter": data["submitter"],
+        "data": {
+            "targets": [
+                {
+                    "name": obj.id,
+                    "ra": obj.ra,
```

Comment:
Not in this one but in a future PR for both Hermes and TNS we should make use to use a ra/dec computed from a flux weighted centroid, just like the `ObjPosition` handler does. That yields a position which is better than the plain ra/dec.